pkg/asset/agent/manifests/agentclusterinstall.go

```diff
@@ -171,5 +190,66 @@ func (a *AgentClusterInstall) finish() error {
 		return errors.New("missing configuration or manifest file")
 	}
 
+	if err := a.validateIPAddressAndNetworkType().ToAggregate(); err != nil {
+		return errors.Wrapf(err, "invalid NetworkType configured")
+	}
+
 	return nil
 }
+
+func (a *AgentClusterInstall) getDefaultNetworkType(installConfig *types.InstallConfig) string {
+	// Returns default network type for a given install config.
+	// The default NetworkType is set to OVNKubernetes if unspecified.
+	defaults.SetInstallConfigDefaults(installConfig)
+	return installConfig.NetworkType
+}
+
+func isIPv4(ipAddress string) bool {
+	return strings.Count(ipAddress, ":") < 2
```

Comment:
What about using instead net.ParseCIDR()?

Reply:
How does one use net.ParseCIDR() to determine if the IP address is v4 or v6?

Reply:
If there's no error, it is possible to reuse the first result from [net.ParseCIDR()](https://pkg.go.dev/net#ParseCIDR) to check if the [IP](https://pkg.go.dev/net#IP) is a v4, ie something like that

```
ip, _, err := net.ParseCIDR(cidrStr)
if err != nil {
  ...
}
ip4 := ip.To4()
if ip4 != nil {
   ... //Ok
}
```
(There are also other nice handy functions in that package to manipulate ip addresses, ie To16() for v6)

Reply:
Unless there is a problem with it, I prefer the current single line implementation.

Reply:
Well, given that the language offers a native library, which is well tested and robust, for that specific topic then IMHO I don't see a reason for not using it. Also, in other code portions of the installer where similar validations are applied the `net` package is tipically adopted

Reply:
Good point. I'll update to use the net package to perform the validation.